src/types/errors.d.ts

```diff
@@ -4,6 +4,7 @@ export type FaunaHttpErrorResponseContent = {
   errors: {
     code: string,
     description: string
+    cause: any;
```

Comment:
The cause field is an array of additional errors, recursively. Also, `position` should also be included, so I think its type can be specified like this

```suggestion
    position: (string | number)[] 
    code: string,
    description: string
    cause?: FaunaHttpErrorResponseContent[];
```

Here's an example raw response from a query where a UDF calls a UDF which then has an error:

```json
{
  "errors": [
    {
      "position": [  ],
      "code": "call error",
      "description": "Calling the function resulted in an error.",
      "cause": [
        {
          "position": [ "expr" ],
          "code": "call error",
          "description": "Calling the function resulted in an error.",
          "cause": [
            {
              "position": [ "expr", "concat", 0 ],
              "code": "invalid argument",
              "description": "String expected, Integer provided."
            },
            {
              "position": [ "expr", "concat", 1 ],
              "code": "invalid argument",
              "description": "String expected, Integer provided."
            }
          ]
        }
      ]
    }
  ]
}
```

Do you have any other examples?

Reply:
I think this is a complete enough example of an error response

It would be better to have a simple query example corresponding to the response, just as the documentation has done

So I think this example can replace the one in the current document, or add an extra example
https://docs.fauna.com/fauna/current/build/fql/errors?lang=javascript